commands/list/clusters/command.go

```diff
@@ -156,40 +175,104 @@ func getClustersOutput(args Arguments) (string, error) {
 		}
 
 		return string(outputBytes), nil
-	} else {
-		if len(response.Payload) == 0 {
-			return color.YellowString("No clusters"), nil
-		}
-		// table headers
-		output := []string{strings.Join([]string{
-			color.CyanString("ID"),
-			color.CyanString("ORGANIZATION"),
-			color.CyanString("NAME"),
-			color.CyanString("RELEASE"),
-			color.CyanString("CREATED"),
-		}, "|")}
-
-		// sort clusters by ID
-		sort.Slice(response.Payload[:], func(i, j int) bool {
-			return response.Payload[i].ID < response.Payload[j].ID
-		})
-
-		for _, cluster := range response.Payload {
-			created := util.ShortDate(util.ParseDate(cluster.CreateDate))
-			releaseVersion := cluster.ReleaseVersion
-			if releaseVersion == "" {
-				releaseVersion = "n/a"
+	}
+
+	// sort clusters by ID
+	sort.Slice(response.Payload[:], func(i, j int) bool {
+		return response.Payload[i].ID < response.Payload[j].ID
+	})
+
+	// This function's output string.
+	output := ""
+
+	if len(response.Payload) == 0 {
+		output += color.YellowString("No clusters")
+	}
+
+	headers := []string{
+		color.CyanString("ID"),
+		color.CyanString("ORGANIZATION"),
+		color.CyanString("NAME"),
+		color.CyanString("RELEASE"),
+		color.CyanString("CREATED"),
+	}
+
+	if args.showDeleted {
+		headers = append(headers, color.CyanString("DELETED"))
+	}
+
+	table := []string{strings.Join(headers, "|")}
+
+	numDeletedClusters := 0
+	numOtherClusters := 0
+
+	for _, cluster := range response.Payload {
+		created := util.ShortDate(util.ParseDate(cluster.CreateDate))
+		deleted := "n/a"
+
+		var secondsSinceDelete float64
+
+		if cluster.DeleteDate != nil {
+			numDeletedClusters++
+
+			if !args.showDeleted {
+				continue
 			}
 
-			output = append(output, strings.Join([]string{
+			deleted = util.ShortDate(util.ParseDate(cluster.DeleteDate.String()))
+			deleteTime := time.Time(*cluster.DeleteDate)
+			secondsSinceDelete = time.Now().Sub(deleteTime).Seconds()
+		} else {
+			numOtherClusters++
+		}
+
+		releaseVersion := cluster.ReleaseVersion
+		if releaseVersion == "" {
+			releaseVersion = "n/a"
+		}
+
+		fields := []string{}
+
+		if secondsSinceDelete > 86400 {
+			fields = []string{
+				color.RedString(cluster.ID),
+				color.RedString(cluster.Owner),
+				color.RedString(cluster.Name),
+				color.RedString(releaseVersion),
+				color.RedString(created),
+			}
+			if args.showDeleted {
+				fields = append(fields, color.RedString(deleted))
+			}
+		} else {
```

Comment:
I would extract repeated logic before the condition but no strong feelings

Reply:
Good point! Changed.